### ISSUE TITLE
`npm ci` fails when `node` dependency does not match system wide install

### ISSUE THREAD

**Issue by setpill** [npm/npm#20957]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [x] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

A project I am working on has a package (zmq) as a dependency that only works with specific versions of node (and the install fails otherwise). It is possible to add the correct node version as a dependency using the `node` package, and this will cause a local install of that specific node version to be made, which might differ from the system-wide version (this is great!).

However, `npm ci` does not handle this properly, while `npm i` from a clean slate does!

#### How can the CLI team reproduce the problem?

1. Have a system with node 10 installed (e.g. docker image `node:latest`).
2. Make sure zmq development headers are installed (when using `node:latest` do `apt-get update && apt-get install -y libzmq3-dev`).
3. Have a project with `"node": "^8.11.2"` and `"zmq": "^2.15.3"` as dependencies in `package.json`.
4. `rm -r node_modules && npm i` -> success.
5. `npm ci` -> failure.

### supporting information:

 - `npm -v` prints:
6.1.0
 - `node -v` prints:
v10.4.0
 - `npm config get registry` prints:
https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?:
Docker image `node:latest`, iow. Debian.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [x] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by legodude17** [comment]
Does this happen if you only have the `"node": "^8.11.2"`? In you `dependencies`?

**Comment by setpill** [comment]
No, because it is the `zmq` package that does not work/breaks during installing under node 10. What seems to happen is that `npm i` installs node 8 first in the `preinstall` phase, but `npm ci` does not. Will see if manually configuring a `preinstall` script is a viable workaround until this is properly fixed.
